src/main/java/com/github/alexdlaird/ngrok/protocol/CreateTunnel.java

```diff
@@ -504,6 +523,9 @@ public void withTunnelDefinition(Map<String, Object> tunnelDefinition) {
             }
             if (isNull(this.schemes) && tunnelDefinition.containsKey("schemes")) {
                 this.basicAuth = (List<String>) tunnelDefinition.get("basic_auth");
+            }
+			if (isNull(this.oauth) && tunnelDefinition.containsKey("oauth")) {
+                this.oauth = (OAuth) tunnelDefinition.get("oauth");
```

Comment:
Is there a test case for this path? I doubt this cast will work, likely needs to go through a constructor or Builder method (that would need to be created).

Reply:
Just checked the ngrok API documentation. OAuth settings are in fact never returned from their /api/tunnels endpoint. I will remove this code.

Reply:
Not quite. The purpose of `tunnelDefinitions` here is that it is read from the config file—anything that can be passed as a construction parameter of a `Tunnel` can also be defined in the `ngrok` config file, you can see the docs for it [here](https://ngrok.com/docs/ngrok-agent/config/#tunnel-definitions). Thus, if `oath` parameters are defined for a tunnel in the config file, it needs to be able to be parsed, and we need to maintain party of Tunnel definition parameters in all places.

[Here is an example](https://github.com/alexdlaird/java-ngrok/blob/a44dd92b8deb340f072d0b122499d090edbda0f9/src/test/java/com/github/alexdlaird/ngrok/NgrokClientTest.java#L707) of starting a tunnel through tunnel definitions in an existing test case.